modulemd/tests/ModulemdTests/modulestream.py

```diff
@@ -1311,6 +1311,93 @@ def test_xmd_issue_290(self):
             self.assertIsNotNone(output_yaml)
         pass
 
+    def test_search_profiles(self):
+        stream = Modulemd.ModuleStreamV2.new("themodule", "thestream")
+
+        # First with no profiles added. Make sure we get back a zero-length
+        # array
+
+        profiles = stream.search_profiles(None)
+        self.assertIsNotNone(profiles)
+        self.assertEqual(len(profiles), 0)
+
+        profiles = stream.search_profiles("*")
+        self.assertIsNotNone(profiles)
+        self.assertEqual(len(profiles), 0)
+
+        profiles = stream.search_profiles("thefirstprofile")
+        self.assertIsNotNone(profiles)
+        self.assertEqual(len(profiles), 0)
+
+        # Now add three profiles, conveniently alphabetical
+        stream.add_profile(Modulemd.Profile.new("thefirstprofile"))
```

Comment:
Might I suggest adding the profiles _not_ in alphabetical order--to demonstrate that `search_profiles()` sorts the returned list as documented.

Reply:
Yes, that's a good idea. I'll do that.